src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -118,16 +119,17 @@ function ReportActionItemMessageEdit(props) {
         });
 
         return () => {
-            // Skip if this is not the focused message so the other edit composer stays focused
-            if (!isFocusedRef.current) {
+            // Skip if this is not the focused message so the other edit composer stays focused.
+            // In small screen devices, when EmojiPicker is shown, the current edit message will lose focus, we need to check this case as well.
+            if (!isFocusedRef.current && !EmojiPickerAction.isActiveReportAction(props.action.reportActionID)) {
```

Comment:
This caused a [regression](https://github.com/Expensify/App/issues/23908) (case 3). If we focus on message A, open it's emoji picker and close it, then focus on message B and delete message A then the main composer will show up. This is due to the faulty `EmojiPickerAction.isActiveReportAction` check as it returns true for message A even though the emoji picker is no longer open.

What we had to do is to clear the active report action accordingly i.e. if we are deleting message A and it's emoji picker is not open then `EmojiPickerAction.isActiveReportAction` should return false.